### PR TITLE
feat: bootstrap peers

### PR DESCRIPTION
**Pull request by AaronGoldman**
bootstrap peers for Mainnet, TestnetClay, and DevUnstable from
https://github.com/ceramicnetwork/
js-ceramic/blob/develop/packages/ipfs-topology/src/ipfs-topology.ts

### CODE REVIEW

File: one/src/main.rs [hunk 1]
```diff
@@ -220,6 +220,40 @@ impl Daemon {
         };
         debug!("using directory: {}", dir.display());
 
+        let mut bootstrap_addresses = opts.bootstrap_addresses.clone();
```
Comment (nathanielc) [inline one/src/main.rs:223]: I believe we decided to remove the `bootstrap_addresses` from the CLI options. Can you make that change?

File: one/src/main.rs [hunk 2]
```diff
@@ -220,6 +220,40 @@ impl Daemon {
...
+        // Add the default bootstrap peers for Mainnet, TestnetClay, and DevUnstable.
+        // from https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/ipfs-topology/src/ipfs-topology.ts
+        let bootstrap_addresses = match opts.network {
```
Comment (nathanielc) [inline one/src/main.rs:226]: We could add this `match` expression as a method on the `Network` type itself. That way its clear the list of bootstrap address is a property of the network selection itself. Thoughts?

File: one/src/main.rs [hunk 3]
```diff
@@ -287,11 +317,14 @@ impl Daemon {
             .build(sql_pool.clone())
             .await?;
 
+        let bind_address = opts.bind_address.clone();
+        let metrics_bind_address = opts.metrics_bind_address.clone();
```
Comment (nathanielc) [inline one/src/main.rs:321]: What is this change for? Why do we clone the address?

**Review by nathanielc — CHANGES_REQUESTED** [review]
LGTM with a few small changes

File: one/src/main.rs (hunk 1 above)
Comment (AaronGoldman) [reply one/src/main.rs]: removed

File: one/src/main.rs (hunk 2 above)
Comment (AaronGoldman) [reply one/src/main.rs]: Moved to Network enum looks cleaner.

File: one/src/main.rs (hunk 3 above)
Comment (AaronGoldman) [reply one/src/main.rs]: Removed

This was a change from a different branch by accident.
 I was changing the way we pass the opts to have all DaemonOpts stored on the Daemon struct.
`daemon.opts` so they could be used in `daemon.run()` not just `daemon.build()`.

File: one/src/main.rs [hunk 4]
```diff
@@ -136,6 +133,34 @@ impl Network {
...
+                "/dns4/go-ipfs-ceramic-private-qa-external.3boxlabs.com/tcp/4011/ws/p2p/QmXcmXfLkkaGbQdj98cgGvHr5gkwJp4r79j9xbJajsoYHr".to_string(),  // cspell:disable-line
+                "/dns4/go-ipfs-ceramic-private-cas-qa-external.3boxlabs.com/tcp/4011/ws/p2p/QmRvJ4HX4N6H26NgtqjoJEUyaDyDRUhGESP1aoyCJE1X1b".to_string(),  // cspell:disable-line
+            ],
+            Network::Local => vec![],
+            Network::InMemory => vec![],
+        }
+        .iter()
+        .map(|addr| addr.parse())
+        .collect::<Result<Vec<Multiaddr>, multiaddr::Error>>()
+        .expect("hard coded bootstrap addresses parse failed")
```
Comment (nathanielc) [inline one/src/main.rs:162]: nit, the idiomatic tense for this is `should`:

```
hard coded bootstrap addresses should parse
```

**Review by nathanielc — APPROVED** [review]
LGTM with one small nit

**Review by nathanielc — APPROVED** [review]
LGTM

**Pull request merged**